### PR TITLE
[setup] Add code for setup.py test to work

### PR DESCRIPTION
**Pull request by jgbarah**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,10 +23,13 @@
 
 import codecs
 import os
+import os.path
```
Comment (sduenas) [inline setup.py]: This isn't needed. `os` import do the same work.

File: setup.py [hunk 2]
```diff
@@ -90,5 +111,8 @@
         'pandas>=0.17',
         'pyyaml>=3.12'
       ],
+      cmdclass = cmdclass,
+#      tests_require=[],
```
Comment (sduenas) [inline setup.py]: Remove this comments.

File: setup.py [hunk 3]
```diff
@@ -49,6 +52,24 @@
...
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        test_suite = unittest.TestLoader().discover('.', pattern='test*.py')
+        result = unittest.TextTestRunner(buffer=True).run(test_suite)
+        sys.exit(not result.wasSuccessful())
+
+cmdclass = {'test':TestCommand}
```
Comment (sduenas) [inline setup.py]: This doesn't follow PEP8 specification.

**Comment by jgbarah** [comment]
Oooops. I thought it was self-explanatory. `setup.py test` was not working, now it works...

If you want me to elaborate, that means that now we can run tests as they are usually run when using the setup.py machinery, and they still run when running `run_tests.py` by hand. I'm going to include some of this in the commit, so that it is more clear.

Thanks for the comments, I will address them asap.

File: setup.py (hunk 1 above)
Comment (jgbarah) [reply setup.py]: You're right, sorry.

File: setup.py (hunk 3 above)
Comment (jgbarah) [reply setup.py]: Fixed. I also fixed some other issues reported by flake8.

File: setup.py (hunk 2 above)
Comment (jgbarah) [reply setup.py]: Done.

**Pull request merged**